templates/terraform/pre_delete/managed_dns_zone.go.erb

```diff
@@ -0,0 +1,56 @@
+if d.Get("force_destroy").(bool) {
+	zone := d.Get("name").(string)
+	token := ""
+	for paginate := true; paginate; {
+		resp, err := config.clientDns.ResourceRecordSets.List(project, zone).PageToken(token).Do()
+		if err != nil {
+			return fmt.Errorf("Error reading ResourceRecordSets: %s", err)
+		}
+
+		for _, rr := range resp.Rrsets {
+			// Build the change
+			chg := &dns.Change{
+				Deletions: []*dns.ResourceRecordSet{
+					{
+						Name:    rr.Name,
+						Type:    rr.Type,
+						Ttl:     rr.Ttl,
+						Rrdatas: rr.Rrdatas,
+					},
+				},
+			}
+
+			if rr.Type == "NS" {
+				mz, err := config.clientDns.ManagedZones.Get(project, zone).Do()
+				if err != nil {
+					return fmt.Errorf("Error retrieving managed zone %q from %q: %s", zone, project, err)
+				}
+				domain := mz.DnsName
+
+				if domain == rr.Name {
```

Comment:
I think you can probably skip all NS records even if the DnsName doesn't match the ResourceRecordSet name.

Reply:
The NS and SOA records can't be deleted.